integration_test/cases/browser/shadow_dom_test.exs

```diff
@@ -0,0 +1,49 @@
+defmodule Wallaby.Integration.Browser.ShadowDomTest do
+  use Wallaby.Integration.SessionCase, async: true
+
+  setup %{session: session} do
+    page =
+      session
+      |> visit("shadow_dom.html")
+
+    {:ok, page: page}
+  end
+
+  test "can find a shadow root", %{session: session} do
+    shadow_root =
+      session
+      |> find(Query.css("shadow-test"))
+      |> shadow_root()
+
+    assert shadow_root
+  end
+
+  test "can find stuff within da shadow dom", %{session: session} do
+    element =
+      session
+      |> find(Query.css("shadow-test"))
+      |> shadow_root()
+      |> find(Query.css("#in-shadow"))
+
+    assert element
```

Comment:
let's actually assert on the contents of these values rather than just on truthiness